xbout/plotting/animate.py

```diff
@@ -7,7 +7,7 @@
 
 
 def animate_imshow(data, animate_over='t', x='x', y='y', animate=True,
```

Comment:
It's likely that we shouldn't actually be using `imshow`, because it's restricted to square images right? Though I originally wrote this for slab cross-sections, then really this should use `pcolormesh` so that it can handle any geometry? Integrating it with the multi-region plotting method used in #34 will add a bit of complexity (though I think the blocks abstraction will mean it's not too hard).

Reply:
Think I agree with all of this, although it does need a bit of thinking... Having poloidal plots with geometry included is (very, very) nice to have as an option (maybe even default), but having cartesian as an option too is also useful - for example poloidal plotting of some arbitrary slice is going to be a bit tricky*; also if there are some very small or distorted cells they may be easier to see in a plot that's just rectangular. The blocks abstraction does very much help with all of this, we can add over time functions that create a block in various ways, and code like this that sticks blocks together would only need to add some logic to say which function to call.

It probably would be sensible to use `pcolormesh` instead of `imshow` everywhere though, since it's much more flexible.

--
* Maybe it would be useful to introduce some 'coordinates' that say which region a point is, e.g. `xregion` where `xregion=0` for `x<ixseps1`, `xregion=1` for `ixseps1<=x<ixseps2`, `xregion=2` for `x>=ixseps2` [note ixseps1/ixseps2 include the x-boundary cells, so need to subtract `MXG` if x does not include the boundary cells]. Hopefully this can be done in such a way that if you do something like `n.isel(x=slice(3,7,1)` then `xregion` would get sliced as well. Then we could select regions to join with `xregion` and a `yregion` (defined using `jyseps*`)?